test/testlib/minikube_utilities.go

```diff
@@ -727,6 +740,15 @@ func getAppLogStream(t *testing.T, namespace string, podName string, podLogOptio
 	client, err := k8s.GetKubernetesClientFromOptionsE(t, options)
 	require.NoError(t, err)
 
+	if podLogOptions.Container == "" {
+		// Select first container if not specified; otherwise the GetLogs method will fail if there are sidecars
```

Comment:
sounds like the expectation is to get logs from the "engine" container, or "admin" container. Do we have the naming convention in place? Can we select either engine or admin, rather than [0]?

Reply:
In practice, the first element is the main container in the pod (either `admin` or `engine`). It's a shame that Golang doesn't have generic slice/array filter or find methods. If we need to explicitly find the container name, we should add something like:
```
var containerName string
for _, container := range pod.Spec.Containers {
  if container.Name == "admin" || container.Name == "engine" {
    containerName = container.Name
    break
  }
}
if containerName == "" {
  containerName = pod.Spec.Containers[0].Name
}
```
I wished that we can avoid that...

Reply:
I would do this: https://play.golang.org/p/Q0UET49Ez6T

Reply:
Okay, I've added another function as suggested.